modules/plugin/jdbc/jdbc-hana/src/main/java/org/geotools/data/hana/HanaDialect.java

```diff
@@ -974,6 +1041,42 @@ private void encodeIdentifiers(StringBuffer sb, String... ids) {
         }
     }
 
+    private HanaVersion queryCloudVersion(Connection cx) throws SQLException {
+        if (hanaVersion.getVersion() < 4)
+            return HanaVersion.INVALID_VERSION;
+
+        Statement st = null;
+        ResultSet rs = null;
+        try {
+            st = cx.createStatement();
+            rs = st.executeQuery("SELECT CLOUD_VERSION FROM SYS.M_DATABASE");
+
+            if (!rs.next())
+                return HanaVersion.INVALID_VERSION;
+
+            String cloudVersion = rs.getString(1).replace('-', '.');
+            String[] parts = cloudVersion.split("\\.");
+            if (parts.length < 3)
+                return HanaVersion.INVALID_VERSION;
+            return new HanaVersion(Integer.parseInt(parts[0]), Integer.parseInt(parts[1]), Integer.parseInt(parts[2]));
+        } finally {
+            dataStore.closeSafe(st);
+            dataStore.closeSafe(rs);
```

Comment:
Close the result set before the statement.

Reply:
Done.